bluepyopt/ephys/simulators.py

```diff
@@ -182,3 +182,161 @@ def __init__(self, message, original):
 
         super(NrnSimulatorException, self).__init__(message)
         self.original = original
+
+
+class LFPySimulator(object):
+
+    """LFPy simulator"""
+
+    def __init__(self, LFPyCellModel, electrode=None, cvode_active=True,
+                 cvode_minstep=None, random123_globalindex=None,
+                 mechanisms_directory=None):
+        """Constructor
+        Args:
+            LFPyCellModel (LFPyCellModel): the LFPy cell model
+            electrode (MEAutility.MEA): the MEAutility probe used to compute
+                extracellular signals
+            cvode_active (bool): should neuron use the variable time step
+                integration method
+            cvode_minstep (float): the minimum time step allowed for a cvode
+                step. Default is 0.0.
+            random123_globalindex (int): used to set the global index used by
+                all instances of the Random123 instances of Random
+            mechanisms_directory (str): path to the parent directory of the
+                directory containing the mod files. If the mod files are in
+                "./data/mechanisms", then mechanisms_directory should be
+                "./data/".
+        """
+
+        self.LFPyCellModel = LFPyCellModel
+        self.electrode = electrode
+        self.effective_electrode = electrode
+
+        self.lfpyelectrode = None
+
+        if platform.system() == "Windows":
+            # hoc.so does not exist on NEURON Windows
+            # although \\hoc.pyd can work here, it gives an error for
+            # nrn_nobanner_ line
+            self.disable_banner = False
+            self.banner_disabled = False
+        else:
+            self.disable_banner = True
+            self.banner_disabled = False
+
+        self.mechanisms_directory = mechanisms_directory
+        self.neuron.h.load_file('stdrun.hoc')
+
+        self.cvode_active = cvode_active
+        self.cvode_minstep_value = cvode_minstep
+
+        self.random123_globalindex = random123_globalindex
+
+    @staticmethod
+    def _nrn_disable_banner():
+        """Disable Neuron banner"""
+
+        nrnpy_path = os.path.join(imp.find_module("neuron")[1])
+        import glob
+
+        hoc_so_list = glob.glob(os.path.join(nrnpy_path, "hoc*.so"))
+
+        if len(hoc_so_list) != 1:
+            warnings.warn(
+                "Unable to find Neuron hoc shared library in %s, "
+                "not disabling banner" % nrnpy_path
+            )
+        else:
+            hoc_so = hoc_so_list[0]
+            nrndll = ctypes.cdll[hoc_so]
+            ctypes.c_int.in_dll(nrndll, "nrn_nobanner_").value = 1
```

Comment:
This is copied paster from the `NrnSimulator` and not used. 
It can be deleted or the static method can become a function to be used in both simulators.

Reply:
👍🏻